configuration/pih/scripts/global/zl.js

```diff
@@ -185,32 +185,35 @@ function setUpObsWithObsDateTime(widgetId){
     }
 }
 
-function returnVisitDateValidation(encounterDate,returnVisitDateLessThanEncounterDateMsg,badReturnVisitDateMsg){
+function setupReturnVisitDateValidation(encounterDate,returnVisitDateLessThanEncounterDateMsg,badReturnVisitDateMsg){
 
-   const domEl= jq("#apptDate").find("input[type=text]");
+   const domEl= getField('apptDate.value');
    const yrRange = encounterDate.getFullYear() + ":" + (new Date().getFullYear() + 1);
-   domEl.datepicker('option','yearRange', yrRange); 
-   domEl.prop("readonly", "readonly"); 
-
-  jq(domEl).change(function(){
-    const nextVisitDate = domEl.datepicker('getDate');
-    const months = (nextVisitDate.getFullYear() - encounterDate.getFullYear()) * 12;
-
-    if(months < 12  && nextVisitDate > encounterDate){
-      jq("#return-visit-date-error-message").text('')
-      jq('.submit').prop("disabled", false);
-    }
-    else if(months == 12 && nextVisitDate.getMonth() <= encounterDate.getMonth()){
-      jq("#return-visit-date-error-message").text('')
-      jq('.submit').prop("disabled", false);
-    }else if(months <= 0 && encounterDate > nextVisitDate){
-        jq("#return-visit-date-error-message").text(returnVisitDateLessThanEncounterDateMsg)
-        jq('.submit').prop("disabled", true);
-    }
-    else{
-      jq("#return-visit-date-error-message").text(badReturnVisitDateMsg)
-      jq('.submit').prop("disabled", true);
-    }
-
-  })        
+   domEl.datepicker('option','yearRange', yrRange);
+   domEl.prop("readonly", "readonly");
+
+   const returnVisitDateValidation = function() {
+     const nextVisitDate = domEl.datepicker('getDate');
+     const months = (nextVisitDate.getFullYear() - encounterDate.getFullYear()) * 12;
+
+     if (months < 12  && nextVisitDate > encounterDate) {
+       getField('apptDate.error').text('').hide();
```

Comment:
This uses the "getField" method (documentation in the link I added above) to hide/set the error text on the "span" with class "error" that HTML Form Entry provides for us.